src/simplify_dyn_ops.cpp

```diff
@@ -319,6 +320,39 @@ struct find_const_alloc_fill
     }
 };
 
+/**
+ * Simplify broadcast_for_dot instructions with two static shaped arguments
+ * From:
+ * broadcast_for_dot(static_shape_arg, static_shape_arg)
+ * To:
+ * multibroadcast(static_shape_arg); output_lens = static_broadcast_for_doted_shape
+ */
+struct find_static_broadcast_for_dot
+{
+    auto matcher() const
+    {
+        return match::name("broadcast_for_dot")(match::arg(0)(match::static_shape()),
+                                                match::arg(1)(match::static_shape()));
+    }
+
+    void apply(module& m, const match::matcher_result& mr) const
+    {
+        auto broadcast_for_dot_ins = mr.result;
+        auto inputs                = broadcast_for_dot_ins->inputs();
+        auto s0                    = inputs.at(0)->get_shape();
+        auto s1                    = inputs.at(1)->get_shape();
```

Comment:
Style: a resize() operation is what you are emulating..

Reply:
I'm not following what you mean here. There's no resize() function for `shape`.

Reply:
As in, l0_broadcasted_lens is simply a resize of s0 lens. l1_broadcasted_lens too.. similar to https://github.com/ROCm/AMDMIGraphX/pull/2629/files/1adde18494a9918d855ce3ffc27f42b848bd89d5#r1500019530